src/sleep/sleep.c

```diff
@@ -349,11 +353,39 @@ static int custom_timer_suspend(const SleepConfig *sleep_config) {
         return 1;
 }
 
+/* Freeze when invoked and thaw on cleanup */
+static int freeze_thaw_user_slice(const char **method) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+        int r;
+
+        if (!method)
```

Comment:
on cleanup *method will be set, but the value it's pointing to will be null, so you need: `if (!method || !*method)`

Reply:
yeah both should be set. nice pointed out.